### ISSUE TITLE
Consider ways to improve the stability of the Unicode rules

### ISSUE THREAD

**Issue by kbenoit** [quanteda/quanteda#1996]
Different platforms have different underlying Unicode versions and versions of the ICU library implemented by **stringi**, and this can result in differences in how the built-in tokenizers handle segmentation.

To enable GitHub Actions CI, which includes Ubuntu 20.04, for instance, we had to skip this test:

https://github.com/quanteda/quanteda/blob/bfe304ac2609c747f35198b98b43c8bf4bd26363/tests/testthat/test-tokens.R#L1116-L1127

None of us currently has Ubuntu 20.04 installed, but my tests on other platforms that we do have revealed this:
My macOS:

```r
> stringi::stri_info()[1:2]
$Unicode.version
[1] "10.0"

$ICU.version
[1] "61.1"
```

Windows 10:
```r
> stringi::stri_info()[1:2]
$Unicode.version
[1] "10.0"

$ICU.version
[1] "61.1"
```

Ubuntu 18.04:
```r
> stringi::stri_info()[1:2]
$Unicode.version
[1] "12.1"

$ICU.version
[1] "64.2"
```

Debian 9:
```r
> stringi::stri_info()[1:2]
$Unicode.version
[1] "8.0"

$ICU.version
[1] "57.1"
```

[Here](https://github.com/quanteda/quanteda/pull/1994#issuecomment-677144071), @koheiw has a good suggestion:

> If the test is breaking on recent ICU, worth investigating. You might remember that the ICU boundary detection changed its handling of a sequence of the white space, and broke many tests while ago.
> 
> Also, we should print the installed version of ICU on start up to make users aware of different outputs.

**Comment by koheiw** [comment]
Only way to solve the problem might be installing stringi with the bundle ICU library.

https://github.com/gagolews/stringi/blob/7e5d5ff616be13dcdea22c901da3ef7af9911ad5/INSTALL#L30-L49

**Comment by koheiw** [comment]
A more radial "solution" is skip all the tests when the ICU is too old (`Unicode.version< 12.0`). We can warn users of potential problem when quanteda is loaded: "The result of analysis could be inaccurate, because the stringi package is built with old Unicode library."

**Comment by kbenoit** [comment]
That would work for tests but we still have the examples breaking on Solaris.

**Comment by koheiw** [comment]
I don't think we cal solve this problem in quanteda alone. We have to ask @gagolews to build stringi with a recent ICU library.

**Comment by gagolews** [comment]
Yes, I am planning to do this later this year.
https://github.com/gagolews/stringi/issues/401

Solaris will still be problematic, as their C++ compiler does not support C++11, so I must keep ICU 55 as a fallback version there. :/

**Comment by koheiw** [comment]
@gagolews that would be great. We should also start petition to drop Solaris from CRAN check...

**Comment by gagolews** [comment]
Good luck with this! :)

Luckily, hardware has limited life-time...

**Comment by kbenoit** [comment]
Implement a check of the ICU/Unicode version before looking for the emoji categories, as this is what is breaking on Solaris. 

Cf. https://github.com/quanteda/quanteda.textstats/issues/24, https://github.com/quanteda/quanteda.textstats/issues/28

**Comment by gagolews** [comment]
FYI: stringi is now shipped with ICU 69.1 #401.
I will submit v1.6.1 to CRAN this week.
Hope it won't break quanteda&co. :cold_sweat:
